### PR TITLE
Changed docopt parsing tool with cxxopts tool

### PR DESCRIPTION
**Pull request by tmanalog**
Signed-off-by: Timotei Molcut <timotei.molcut@analog.com>
Docopt parsing tool is not good for scaling up, because when adding new arguments, the stack size must be increased. Cxxopts seems to be a better solution.

### CODE REVIEW

File: examples/data_collect/main.cpp [hunk 1]
```diff
@@ -749,7 +717,7 @@ void fileWriterTask( const thread_params * const pThreadParams ) {
 
     char out_file[MAX_FILE_PATH_SIZE] = {0};
     if (nullptr == pThreadParams->pFsfParams) {
-        snprintf(out_file, sizeof(out_file), "%s/%s_frame_%s_%05lld.bin", pThreadParams->pFolderPath, pThreadParams->pFrame_type, pThreadParams->nFileTime,
```
Comment (tmanalog) [inline examples/data_collect/main.cpp:752]: This line generated a format warning, so I changed the format type.

**Comment by dNechita** [comment]
Let's move the header to a location that could be shared by any other command line executable.
I propose to create a 'common' directory in the top-level directory. Then create a 'third_party' dir inside common. Then I would add this library as a submodule inside 'third_party' dir.

**Comment by dNechita** [comment]
Let's move the header to a location that could be shared by any other command line executable.
I propose to create a 'common' directory in the top-level directory. Then create a 'third_party' dir inside common. Then I would add this library as a submodule inside 'third_party' dir.

**Comment by dNechita** [comment]
With cxxopts library we won't be able to keep the same names for all options. Specifically, we won't be able to have one character length for long options (e.g. --n, --m).
We will put this PR on hold for now.

**Pull request closed without merging**